File: server/game_core/command.py
class Command(object):
    targetID = None
    action = None
    args = None

    def __init__(self, targetid, action, args=[]):
        self.targetID = targetid
        self.action = action
        self.args = args

    def execute(self, target):
        getattr(target, self.action)(self.args)

    def serialize(self):
        return [self.targetID, self.action, self.args]

File: src/client/lib/game/core/command.dart
import 'dart:mirrors';
import 'package:client/game/core/abstract/game_object.dart';
import 'dart:convert';

class Command
{
  String targetID;
  String action;
  List args;

  Command(this.targetID, this.action, [this.args = const []]){}

  Command.from(List serialization)
  {
    targetID = serialization[0];
    action = serialization[1];
    args = serialization[2];
  }

  execute(GameObject target)
  {
    //Invoke a command on a given GameObject
    InstanceMirror mirror = reflect(target);
    mirror.invoke(MirrorSystem.getSymbol(action), args);
  }

  String serialize()
  {
    return JSON.encode([targetID, action,  args]);
  }
}